src/main/java/org/galatea/starter/entrypoint/IexRestController.java

```diff
@@ -48,4 +64,52 @@ public List<IexLastTradedPrice> getLastTradedPrice(
     return iexService.getLastTradedPriceForSymbols(symbols);
   }
 
+  /**
+   * Get the historical prices of specified symbols.
+   *
+   * @param symbols list of symbols to get historical price for.
+   * @param range Optional Parameter specifying the Range of time that the price should be
+   *     measured over
+   * @param date Optional Parameter specifying the exact date the price should be measured on
+   * @return a IexLastTradedPrice object for the given symbol.
+   */
+  @GetMapping(value = "${mvc.iex.getHistoricalPricePath}", produces = {
+      MediaType.APPLICATION_JSON_VALUE})
+  public IexHistoricalPrice getHistoricalPrice(
+      @RequestParam(value = "symbol") @NotNull final String symbols,
+      @RequestParam(value = "range", required = false) final Optional<@Pattern.List(
+          {@Pattern(regexp = ".*"), @Pattern(regexp = VALID_DATES)}) String> range,
+      @RequestParam(value = "date", required = false) final Optional<Integer> date) {
+    IexHistoricalPrice price;
+//@Pattern.List({@Pattern(regexp =".*"), @Pattern (regexp = VALID_DATES)})
+    //3 Parameters: Symbol is REQUIRED, Date/Range are optional and cannot both be present
+
+    if (date.isPresent()) {
+      int dateVal = date.get();
+
+      //Max validation depends on current time:
+      if (dateVal >= System.currentTimeMillis()) {
+        throw new IllegalArgumentException("Cannot get a price for a time in the future");
+      }
+
+      //Min validation depends on current time:
+      long fiveYearsAgo = Long.parseLong(LocalDate.now().toString().replace("-", ""));
```

Comment:
Try comparing the input date to now() using some built-in comparisons within the Java 8 time APIs, rather than converting to long.
Also, using `LocalDate.now()` can make testing more difficult, since the value of now() will change from day to day. See if you can find a way to optionally mock out this value, so that you can get proper behavior (i.e. the actual "now" time) when running normally but some predefined mock value during tests.